sql-plugin/src/main/311until320-nondb/scala/com/nvidia/spark/rapids/shims/v2/SparkBaseShims.scala

```diff
@@ -165,7 +165,7 @@ abstract class SparkBaseShims extends Spark31XShims {
 
         // stringChecks are the same
         // binaryChecks are the same
-        override val decimalChecks: TypeSig = DECIMAL_64 + STRING
+        override val decimalChecks: TypeSig = gpuNumeric + STRING
```

Comment:
It looks like the shims are overriding this to the same behavior as the base method.  Would it be a better change to stop overriding the base version in the shims that don't need to change the behavior, or was this intentionally overridden redundantly?

Reply:
Some of this was done because Spark itself changed what could be cast (dropped some things that just returned nulls). So at a minimum we had to override `sparkDecimalSig` but I thought it was cleaner to keep it all together and included `decimalChecks` with it.